usr/src/cmd/fs.d/nfs/statd/sm_svc.c

```diff
@@ -457,6 +460,22 @@ thr_statd_merges(void)
 	(void) mutex_unlock(&merges_lock);
 }
 
+static int
+convert_int(int *val, char *str)
+{
+	long lval;
+
+	if (str == NULL || !isdigit(*str))
+		return (-1);
+
+	lval = strtol(str, &str, 10);
```

Comment:
The error checking here is incomplete. You need to initialize errno to zero before calling this and also check that errno is non-zero. It's possible for strtol() to return zero and set errno to EINVAL.